libs/vkd3d-shader/spirv.c

```diff
@@ -4486,6 +4486,41 @@ static unsigned int vkd3d_dxbc_compiler_get_output_variable_index(
     return register_idx;
 }
 
+static DWORD vkd3d_dxbc_compiler_double_source_mask_fixup(enum vkd3d_data_type src, enum vkd3d_data_type dst, DWORD mask)
+{
+    /* In DXBC, doubles take up a .xy whereas a normal float/int
+     * takes up a .x or a .y, etc.
+     * This maps a .xy writemask to a .x when going between
+     * double -> float/int and vice-versa for float/int -> double.
+     */
+
+    if (src == dst)
```

Comment:
kind of redundant

Reply:
Not when we call this in regular movc which could have this.

You could theoretically insert this into any normal function.

Reply:
Or any ops that have DOUBLE == DOUBLE.

The alternative here would be SRC == DOUBLE && DST != DOUBLE and vice versa.

Reply:
The point is that we take the same path anyway when falling through the `if`s below.

Reply:
That's not true when src == DOUBLE and dst == DOUBLE